heudiconv/dicoms.py

```diff
@@ -138,135 +236,72 @@ def group_dicoms_into_seqinfos(files, file_filter, dcmfilter, grouping):
+        if mw.image_shape is None:
+            # this whole thing has no image data (maybe just PSg DICOMs)
+            # nothing to see here, just move on
+            continue
+        seqinfo = create_seqinfo(mw, series_files, series_id)
 
-        size = list(mw.image_shape) + [len(series_files)]
-        total += size[-1]
-        if len(size) < 4:
-            size.append(1)
-
-        # MG - refactor into util function
-        try:
-            TR = float(dcminfo.RepetitionTime) / 1000.
-        except (AttributeError, ValueError):
-            TR = -1
-        try:
-            TE = float(dcminfo.EchoTime)
-        except (AttributeError, ValueError):
-            TE = -1
-        try:
-            refphys = str(dcminfo.ReferringPhysicianName)
-        except AttributeError:
-            refphys = ''
-        try:
-            image_type = tuple(dcminfo.ImageType)
-        except AttributeError:
-            image_type = ''
-        try:
-            series_desc = dcminfo.SeriesDescription
-        except AttributeError:
-            series_desc = ''
-
-        motion_corrected = 'MOCO' in image_type
-
-        if dcminfo.get([0x18,0x24], None):
-            # GE and Philips scanners
-            sequence_name = dcminfo[0x18,0x24].value
-        elif dcminfo.get([0x19, 0x109c], None):
-            # Siemens scanners
-            sequence_name = dcminfo[0x19, 0x109c].value
-        else:
-            sequence_name = 'Not found'
-
-        info = SeqInfo(
-            total,
-            op.split(series_files[0])[1],
-            series_id,
-            op.basename(op.dirname(series_files[0])),
-            '-', '-',
-            size[0], size[1], size[2], size[3],
-            TR, TE,
-            dcminfo.ProtocolName,
-            motion_corrected,
-            'derived' in [x.lower() for x in dcminfo.get('ImageType', [])],
-            dcminfo.get('PatientID'),
-            dcminfo.get('StudyDescription'),
-            refphys,
-            series_desc,  # We try to set this further up.
-            sequence_name,
-            image_type,
-            accession_number,
-            # For demographics to populate BIDS participants.tsv
-            dcminfo.get('PatientAge'),
-            dcminfo.get('PatientSex'),
-            dcminfo.get('AcquisitionDate'),
-            dcminfo.get('SeriesInstanceUID')
-        )
-        # candidates
-        # dcminfo.AccessionNumber
-        #   len(dcminfo.ReferencedImageSequence)
-        #   len(dcminfo.SourceImageSequence)
-        # FOR demographics
         if per_studyUID:
             key = studyUID.split('.')[-1]
-        elif per_accession_number:
-            key = accession_number
+        elif grouping == 'accession_number':
+            key = accession_number = mw.dcm_data.get("AccessionNumber")
+        elif grouping == 'all':
+            key = 'all'
+        elif custom_grouping:
+            key = custom = mw.dcm_data.get(custom_grouping)
```

Comment:
it seems that `custom` is used only as a `key`down in the code, i.e. `key` wouldn't change.  So should there be no `custom` and just `key`?